stdlib/system/src/test/java/org/ballerinalang/stdlib/system/FileSystemTest.java

```diff
@@ -0,0 +1,383 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.stdlib.system;
+
+import org.apache.commons.io.FileUtils;
+import org.ballerinalang.launcher.util.BCompileUtil;
+import org.ballerinalang.launcher.util.BRunUtil;
+import org.ballerinalang.launcher.util.CompileResult;
+import org.ballerinalang.model.types.BObjectType;
+import org.ballerinalang.model.values.BBoolean;
+import org.ballerinalang.model.values.BError;
+import org.ballerinalang.model.values.BInteger;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.AssertJUnit.assertNull;
+
+/**
+ * Test class for file system package.
+ *
+ * @since 0.995.0
+ */
+public class FileSystemTest {
+
+    private CompileResult compileResult;
+    private static final String TEMP_DIR = System.getProperty("java.io.tmpdir");
+    private Path tempDirPath;
+    private static final Logger log = LoggerFactory.getLogger(FileSystemTest.class);
+
+    @BeforeClass
+    public void setup() throws IOException {
+        compileResult = BCompileUtil.compile(Paths.get("test-src", "file-system-test.bal").toString());
+        tempDirPath = Paths.get(TEMP_DIR, "data-files");
+        if (Files.notExists(tempDirPath)) {
+            Files.createDirectory(tempDirPath);
+        }
+    }
+
+    @Test(description = "Test for retrieving temporary directory")
+    public void testTempDir() {
+        BValue[] returns = BRunUtil.invoke(compileResult, "testGetTempDir");
+        Assert.assertTrue(returns[0] instanceof BString);
+        String expectedValue = System.getProperty("java.io.tmpdir");
+        Assert.assertEquals(returns[0].stringValue(), expectedValue);
+    }
+
+    @Test(description = "Test for changing file path")
+    public void testFileRename() throws IOException {
+        Path tempSourcePath = null;
+        Path tempDestPath = null;
+        try {
+            tempSourcePath = tempDirPath.resolve("src-file.txt");
```

Comment:
`src-file.txt`, `dest-file.txt` seems to be used in many places. So better to use a constant variable. WDYT?